File: quicksort1.py
import math
import os
import random
import re
import sys

#
# Complete the 'quickSort' function below.
#
# The function is expected to return an INTEGER_ARRAY.
# The function accepts INTEGER_ARRAY arr as parameter.
#

def quickSort(arr):
    # Write your code here
    pivot = arr[0]
    left, equal, right = [], [], []
    
    # dividing elements into correct partitions
    for elem in arr:
        if elem < pivot:
            left.append(elem)
        elif elem == pivot:
            equal.append(elem)
        else:
            right.append(elem)
    
    return left + equal + right

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input("Input the number of elements: ").strip())

    arr = list(map(int, input("Input whitespace-separated elements: ").rstrip().split()))

    result = quickSort(arr)

    fptr.write(' '.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: quicksort1.php
<?php

/*
 * Complete the 'quickSort' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function quickSort($arr) {
    // Write your code here
    $pivot = $arr[0];
    $left = $equal = $right = [];
    
    foreach ($arr as $elem) {
        if ($elem < $pivot) {
            array_push($left, $elem);
        } elseif ($elem == $pivot) {
            array_push($equal, $elem);
        } else {
            array_push($right, $elem);
        }
    }
    // echo "\nLeft : " . implode(", ", $left);
    // echo "\nEqual : " . implode(", ", $equal);
    // echo "\nRight : " . implode(", ", $right);
    
    return array_merge($left, $equal, $right);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = quickSort($arr);

fwrite($fptr, implode(" ", $result) . "\n");

fclose($fptr);
